diff --git a/packages/http-client-java/emitter/src/code-model-builder.ts b/packages/http-client-java/emitter/src/code-model-builder.ts
index 5375778e0..917a4a957 100644
--- a/packages/http-client-java/emitter/src/code-model-builder.ts
+++ b/packages/http-client-java/emitter/src/code-model-builder.ts
@@ -241,7 +241,7 @@ export class CodeModelBuilder {
   }
 
   public async build(): Promise<CodeModel> {
-    this.sdkContext = await createSdkContext(this.emitterContext, "@typespec/http-client-java", {
+    this.sdkContext = await createSdkContext(this.emitterContext, "@azure-tools/typespec-java", {
       versioning: { previewStringRegex: /$/ },
     }); // include all versions and do the filter by ourselves
 
diff --git a/packages/http-client-java/emitter/src/emitter.ts b/packages/http-client-java/emitter/src/emitter.ts
index 317e746d2..848b46275 100644
--- a/packages/http-client-java/emitter/src/emitter.ts
+++ b/packages/http-client-java/emitter/src/emitter.ts
@@ -98,7 +98,7 @@ const EmitterOptionsSchema: JSONSchemaType<EmitterOptions> = {
 };
 
 export const $lib = createTypeSpecLibrary({
-  name: "@typespec/http-client-java",
+  name: "@azure-tools/typespec-java",
   diagnostics: {},
   emitter: {
     options: EmitterOptionsSchema,
diff --git a/packages/http-client-java/generator/http-client-generator-core/pom.xml b/packages/http-client-java/generator/http-client-generator-core/pom.xml
index afa83e5c4..db70c748e 100644
--- a/packages/http-client-java/generator/http-client-generator-core/pom.xml
+++ b/packages/http-client-java/generator/http-client-generator-core/pom.xml
@@ -21,6 +21,11 @@
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>com.azure.tools</groupId>
+      <artifactId>azure-autorest-customization</artifactId>
+      <version>1.0.0-beta.8</version>
+    </dependency>
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core</artifactId>
diff --git a/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/postprocessor/Postprocessor.java b/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/postprocessor/Postprocessor.java
index d44ce8ec4..7953870bc 100644
--- a/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/postprocessor/Postprocessor.java
+++ b/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/postprocessor/Postprocessor.java
@@ -3,9 +3,9 @@
 
 package com.microsoft.typespec.http.client.generator.core.postprocessor;
 
+import com.azure.autorest.customization.Customization;
+import com.azure.autorest.customization.implementation.Utils;
 import com.azure.json.JsonReader;
-import com.microsoft.typespec.http.client.generator.core.customization.Customization;
-import com.microsoft.typespec.http.client.generator.core.customization.implementation.Utils;
 import com.microsoft.typespec.http.client.generator.core.extension.base.util.FileUtils;
 import com.microsoft.typespec.http.client.generator.core.extension.plugin.JavaSettings;
 import com.microsoft.typespec.http.client.generator.core.extension.plugin.NewPlugin;
diff --git a/packages/http-client-java/generator/http-client-generator/src/main/resources/readme/pom.xml b/packages/http-client-java/generator/http-client-generator/src/main/resources/readme/pom.xml
index c99896650..358938595 100644
--- a/packages/http-client-java/generator/http-client-generator/src/main/resources/readme/pom.xml
+++ b/packages/http-client-java/generator/http-client-generator/src/main/resources/readme/pom.xml
@@ -16,9 +16,9 @@
 
   <dependencies>
     <dependency>
-      <groupId>com.microsoft.typespec</groupId>
-      <artifactId>http-client-generator-core</artifactId>
-      <version>1.0.0-beta.1</version>
+      <groupId>com.azure.tools</groupId>
+      <artifactId>azure-autorest-customization</artifactId>
+      <version>1.0.0-beta.8</version>
     </dependency>
   </dependencies>
 
