diff --git a/packages/http-client-java/emitter/src/lib.ts b/packages/http-client-java/emitter/src/lib.ts
index 2d9134c3d..c9ef91f87 100644
--- a/packages/http-client-java/emitter/src/lib.ts
+++ b/packages/http-client-java/emitter/src/lib.ts
@@ -1,6 +1,6 @@
 import { createTypeSpecLibrary, JSONSchemaType, paramMessage } from "@typespec/compiler";
 
-export const LIB_NAME = "@typespec/http-client-java";
+export const LIB_NAME = "@azure-tools/typespec-java";
 
 export interface DevOptions {
   "generate-code-model"?: boolean;
diff --git a/packages/http-client-java/generator/http-client-generator-core/pom.xml b/packages/http-client-java/generator/http-client-generator-core/pom.xml
index ce40ac91d..c3ee24c92 100644
--- a/packages/http-client-java/generator/http-client-generator-core/pom.xml
+++ b/packages/http-client-java/generator/http-client-generator-core/pom.xml
@@ -21,6 +21,11 @@
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>com.azure.tools</groupId>
+      <artifactId>azure-autorest-customization</artifactId>
+      <version>1.0.0-beta.9</version>
+    </dependency>
     <dependency>
       <groupId>com.azure</groupId>
       <artifactId>azure-core</artifactId>
diff --git a/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/postprocessor/Postprocessor.java b/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/postprocessor/Postprocessor.java
index d44ce8ec4..7953870bc 100644
--- a/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/postprocessor/Postprocessor.java
+++ b/packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/postprocessor/Postprocessor.java
@@ -3,9 +3,9 @@
 
 package com.microsoft.typespec.http.client.generator.core.postprocessor;
 
+import com.azure.autorest.customization.Customization;
+import com.azure.autorest.customization.implementation.Utils;
 import com.azure.json.JsonReader;
-import com.microsoft.typespec.http.client.generator.core.customization.Customization;
-import com.microsoft.typespec.http.client.generator.core.customization.implementation.Utils;
 import com.microsoft.typespec.http.client.generator.core.extension.base.util.FileUtils;
 import com.microsoft.typespec.http.client.generator.core.extension.plugin.JavaSettings;
 import com.microsoft.typespec.http.client.generator.core.extension.plugin.NewPlugin;
diff --git a/packages/http-client-java/generator/http-client-generator/src/main/resources/readme/pom.xml b/packages/http-client-java/generator/http-client-generator/src/main/resources/readme/pom.xml
index c99896650..fef2a4caf 100644
--- a/packages/http-client-java/generator/http-client-generator/src/main/resources/readme/pom.xml
+++ b/packages/http-client-java/generator/http-client-generator/src/main/resources/readme/pom.xml
@@ -16,9 +16,9 @@
 
   <dependencies>
     <dependency>
-      <groupId>com.microsoft.typespec</groupId>
-      <artifactId>http-client-generator-core</artifactId>
-      <version>1.0.0-beta.1</version>
+      <groupId>com.azure.tools</groupId>
+      <artifactId>azure-autorest-customization</artifactId>
+      <version>1.0.0-beta.9</version>
     </dependency>
   </dependencies>
 
