From e2cb51c007dd035bbd712acfc4c186d4a56b4383 Mon Sep 17 00:00:00 2001
From: Weidong Xu <weidxu@microsoft.com>
Date: Thu, 7 Aug 2025 13:12:56 +0800
Subject: [PATCH] revert impl in onlineexperimentation

---
 .../OnlineExperimentationClientImpl.java        | 10 ++++++----
 ...lineExperimentationWorkspacesClientImpl.java | 17 ++++++++++-------
 2 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/sdk/onlineexperimentation/azure-analytics-onlineexperimentation/src/main/java/com/azure/analytics/onlineexperimentation/implementation/OnlineExperimentationClientImpl.java b/sdk/onlineexperimentation/azure-analytics-onlineexperimentation/src/main/java/com/azure/analytics/onlineexperimentation/implementation/OnlineExperimentationClientImpl.java
index 32e6622cbc6..ed26312e71d 100644
--- a/sdk/onlineexperimentation/azure-analytics-onlineexperimentation/src/main/java/com/azure/analytics/onlineexperimentation/implementation/OnlineExperimentationClientImpl.java
+++ b/sdk/onlineexperimentation/azure-analytics-onlineexperimentation/src/main/java/com/azure/analytics/onlineexperimentation/implementation/OnlineExperimentationClientImpl.java
@@ -233,7 +233,7 @@ public final class OnlineExperimentationClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<Void>> deleteMetric(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("experimentMetricId") String experimentMetricId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Delete("/experiment-metrics/{experimentMetricId}")
         @ExpectedResponses({ 204 })
@@ -243,7 +243,7 @@ public final class OnlineExperimentationClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Response<Void> deleteMetricSync(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("experimentMetricId") String experimentMetricId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Get("/experiment-metrics")
         @ExpectedResponses({ 200 })
@@ -698,8 +698,9 @@ public final class OnlineExperimentationClientImpl {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Void>> deleteMetricWithResponseAsync(String experimentMetricId,
         RequestOptions requestOptions) {
+        final String accept = "application/json";
         return FluxUtil.withContext(context -> service.deleteMetric(this.getEndpoint(),
-            this.getServiceVersion().getVersion(), experimentMetricId, requestOptions, context));
+            this.getServiceVersion().getVersion(), experimentMetricId, accept, requestOptions, context));
     }
 
     /**
@@ -730,8 +731,9 @@ public final class OnlineExperimentationClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> deleteMetricWithResponse(String experimentMetricId, RequestOptions requestOptions) {
+        final String accept = "application/json";
         return service.deleteMetricSync(this.getEndpoint(), this.getServiceVersion().getVersion(), experimentMetricId,
-            requestOptions, Context.NONE);
+            accept, requestOptions, Context.NONE);
     }
 
     /**
diff --git a/sdk/onlineexperimentation/azure-resourcemanager-onlineexperimentation/src/main/java/com/azure/resourcemanager/onlineexperimentation/implementation/OnlineExperimentationWorkspacesClientImpl.java b/sdk/onlineexperimentation/azure-resourcemanager-onlineexperimentation/src/main/java/com/azure/resourcemanager/onlineexperimentation/implementation/OnlineExperimentationWorkspacesClientImpl.java
index fcacbc0d538..f610d3e49f3 100644
--- a/sdk/onlineexperimentation/azure-resourcemanager-onlineexperimentation/src/main/java/com/azure/resourcemanager/onlineexperimentation/implementation/OnlineExperimentationWorkspacesClientImpl.java
+++ b/sdk/onlineexperimentation/azure-resourcemanager-onlineexperimentation/src/main/java/com/azure/resourcemanager/onlineexperimentation/implementation/OnlineExperimentationWorkspacesClientImpl.java
@@ -128,23 +128,23 @@ public final class OnlineExperimentationWorkspacesClientImpl implements OnlineEx
             @HeaderParam("Content-Type") String contentType, @HeaderParam("Accept") String accept,
             @BodyParam("application/json") OnlineExperimentationWorkspacePatch properties, Context context);
 
-        @Headers({ "Accept: application/json;q=0.9", "Content-Type: application/json" })
+        @Headers({ "Content-Type: application/json" })
         @Delete("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OnlineExperimentation/workspaces/{workspaceName}")
         @ExpectedResponses({ 202, 204 })
         @UnexpectedResponseExceptionType(ManagementException.class)
         Mono<Response<Flux<ByteBuffer>>> delete(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("subscriptionId") String subscriptionId,
             @PathParam("resourceGroupName") String resourceGroupName, @PathParam("workspaceName") String workspaceName,
-            Context context);
+            @HeaderParam("Accept") String accept, Context context);
 
-        @Headers({ "Accept: application/json;q=0.9", "Content-Type: application/json" })
+        @Headers({ "Content-Type: application/json" })
         @Delete("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OnlineExperimentation/workspaces/{workspaceName}")
         @ExpectedResponses({ 202, 204 })
         @UnexpectedResponseExceptionType(ManagementException.class)
         Response<BinaryData> deleteSync(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("subscriptionId") String subscriptionId,
             @PathParam("resourceGroupName") String resourceGroupName, @PathParam("workspaceName") String workspaceName,
-            Context context);
+            @HeaderParam("Accept") String accept, Context context);
 
         @Headers({ "Content-Type: application/json" })
         @Get("/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.OnlineExperimentation/workspaces")
@@ -847,9 +847,10 @@ public final class OnlineExperimentationWorkspacesClientImpl implements OnlineEx
         if (workspaceName == null) {
             return Mono.error(new IllegalArgumentException("Parameter workspaceName is required and cannot be null."));
         }
+        final String accept = "application/json";
         return FluxUtil
             .withContext(context -> service.delete(this.client.getEndpoint(), this.client.getApiVersion(),
-                this.client.getSubscriptionId(), resourceGroupName, workspaceName, context))
+                this.client.getSubscriptionId(), resourceGroupName, workspaceName, accept, context))
             .contextWrite(context -> context.putAll(FluxUtil.toReactorContext(this.client.getContext()).readOnly()));
     }
 
@@ -883,8 +884,9 @@ public final class OnlineExperimentationWorkspacesClientImpl implements OnlineEx
             throw LOGGER.atError()
                 .log(new IllegalArgumentException("Parameter workspaceName is required and cannot be null."));
         }
+        final String accept = "application/json";
         return service.deleteSync(this.client.getEndpoint(), this.client.getApiVersion(),
-            this.client.getSubscriptionId(), resourceGroupName, workspaceName, Context.NONE);
+            this.client.getSubscriptionId(), resourceGroupName, workspaceName, accept, Context.NONE);
     }
 
     /**
@@ -918,8 +920,9 @@ public final class OnlineExperimentationWorkspacesClientImpl implements OnlineEx
             throw LOGGER.atError()
                 .log(new IllegalArgumentException("Parameter workspaceName is required and cannot be null."));
         }
+        final String accept = "application/json";
         return service.deleteSync(this.client.getEndpoint(), this.client.getApiVersion(),
-            this.client.getSubscriptionId(), resourceGroupName, workspaceName, context);
+            this.client.getSubscriptionId(), resourceGroupName, workspaceName, accept, context);
     }
 
     /**
-- 
2.50.1.windows.1

