From 3bb4437a6b964156f53b2e8ed22265e14f056a31 Mon Sep 17 00:00:00 2001
From: Weidong Xu <weidxu@microsoft.com>
Date: Thu, 7 Aug 2025 11:36:55 +0800
Subject: [PATCH] revert impl in keyvault

---
 .../keyvault/keys/implementation/KeyClientImpl.java    | 10 ++++++----
 3 files changed, 18 insertions(+), 12 deletions(-)

diff --git a/sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/implementation/KeyClientImpl.java b/sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/implementation/KeyClientImpl.java
index e8f3f1f2712..559c4022523 100644
--- a/sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/implementation/KeyClientImpl.java
+++ b/sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/implementation/KeyClientImpl.java
@@ -583,7 +583,7 @@ public final class KeyClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<Void>> purgeDeletedKey(@HostParam("vaultBaseUrl") String vaultBaseUrl,
             @QueryParam("api-version") String apiVersion, @PathParam("key-name") String keyName,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Delete("/deletedkeys/{key-name}")
         @ExpectedResponses({ 204 })
@@ -593,7 +593,7 @@ public final class KeyClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Response<Void> purgeDeletedKeySync(@HostParam("vaultBaseUrl") String vaultBaseUrl,
             @QueryParam("api-version") String apiVersion, @PathParam("key-name") String keyName,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Post("/deletedkeys/{key-name}/recover")
         @ExpectedResponses({ 200 })
@@ -3921,8 +3921,9 @@ public final class KeyClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Void>> purgeDeletedKeyWithResponseAsync(String keyName, RequestOptions requestOptions) {
+        final String accept = "application/json";
         return FluxUtil.withContext(context -> service.purgeDeletedKey(this.getVaultBaseUrl(),
-            this.getServiceVersion().getVersion(), keyName, requestOptions, context));
+            this.getServiceVersion().getVersion(), keyName, accept, requestOptions, context));
     }
 
     /**
@@ -3942,8 +3943,9 @@ public final class KeyClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> purgeDeletedKeyWithResponse(String keyName, RequestOptions requestOptions) {
+        final String accept = "application/json";
         return service.purgeDeletedKeySync(this.getVaultBaseUrl(), this.getServiceVersion().getVersion(), keyName,
-            requestOptions, Context.NONE);
+            accept, requestOptions, Context.NONE);
     }
 
     /**
-- 
2.50.1.windows.1

