From c8fd94a2e03f6402b90557f58768384a460f61f3 Mon Sep 17 00:00:00 2001
From: Weidong Xu <weidxu@microsoft.com>
Date: Thu, 7 Aug 2025 13:31:11 +0800
Subject: [PATCH] revert impl in loadtesting

---
 .../LoadTestAdministrationClientImpl.java     | 34 ++++++++++++-------
 .../implementation/LoadTestRunClientImpl.java | 20 ++++++-----
 2 files changed, 33 insertions(+), 21 deletions(-)

diff --git a/sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/implementation/LoadTestAdministrationClientImpl.java b/sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/implementation/LoadTestAdministrationClientImpl.java
index 297869a5e79..255296fc444 100644
--- a/sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/implementation/LoadTestAdministrationClientImpl.java
+++ b/sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/implementation/LoadTestAdministrationClientImpl.java
@@ -378,7 +378,8 @@ public final class LoadTestAdministrationClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<Void>> deleteTestFile(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testId") String testId,
-            @PathParam("fileName") String fileName, RequestOptions requestOptions, Context context);
+            @PathParam("fileName") String fileName, @HeaderParam("Accept") String accept, RequestOptions requestOptions,
+            Context context);
 
         @Delete("/tests/{testId}/files/{fileName}")
         @ExpectedResponses({ 204 })
@@ -388,7 +389,8 @@ public final class LoadTestAdministrationClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Response<Void> deleteTestFileSync(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testId") String testId,
-            @PathParam("fileName") String fileName, RequestOptions requestOptions, Context context);
+            @PathParam("fileName") String fileName, @HeaderParam("Accept") String accept, RequestOptions requestOptions,
+            Context context);
 
         @Delete("/tests/{testId}")
         @ExpectedResponses({ 204 })
@@ -398,7 +400,7 @@ public final class LoadTestAdministrationClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<Void>> deleteTest(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testId") String testId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Delete("/tests/{testId}")
         @ExpectedResponses({ 204 })
@@ -408,7 +410,7 @@ public final class LoadTestAdministrationClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Response<Void> deleteTestSync(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testId") String testId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Patch("/test-profiles/{testProfileId}")
         @ExpectedResponses({ 200, 201 })
@@ -440,7 +442,7 @@ public final class LoadTestAdministrationClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<Void>> deleteTestProfile(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testProfileId") String testProfileId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Delete("/test-profiles/{testProfileId}")
         @ExpectedResponses({ 204 })
@@ -450,7 +452,7 @@ public final class LoadTestAdministrationClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Response<Void> deleteTestProfileSync(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testProfileId") String testProfileId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Get("/test-profiles/{testProfileId}")
         @ExpectedResponses({ 200 })
@@ -2730,8 +2732,9 @@ public final class LoadTestAdministrationClientImpl {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Void>> deleteTestFileWithResponseAsync(String testId, String fileName,
         RequestOptions requestOptions) {
+        final String accept = "application/json";
         return FluxUtil.withContext(context -> service.deleteTestFile(this.getEndpoint(),
-            this.getServiceVersion().getVersion(), testId, fileName, requestOptions, context));
+            this.getServiceVersion().getVersion(), testId, fileName, accept, requestOptions, context));
     }
 
     /**
@@ -2749,8 +2752,9 @@ public final class LoadTestAdministrationClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> deleteTestFileWithResponse(String testId, String fileName, RequestOptions requestOptions) {
+        final String accept = "application/json";
         return service.deleteTestFileSync(this.getEndpoint(), this.getServiceVersion().getVersion(), testId, fileName,
-            requestOptions, Context.NONE);
+            accept, requestOptions, Context.NONE);
     }
 
     /**
@@ -2767,8 +2771,9 @@ public final class LoadTestAdministrationClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Void>> deleteTestWithResponseAsync(String testId, RequestOptions requestOptions) {
+        final String accept = "application/json";
         return FluxUtil.withContext(context -> service.deleteTest(this.getEndpoint(),
-            this.getServiceVersion().getVersion(), testId, requestOptions, context));
+            this.getServiceVersion().getVersion(), testId, accept, requestOptions, context));
     }
 
     /**
@@ -2785,8 +2790,9 @@ public final class LoadTestAdministrationClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> deleteTestWithResponse(String testId, RequestOptions requestOptions) {
-        return service.deleteTestSync(this.getEndpoint(), this.getServiceVersion().getVersion(), testId, requestOptions,
-            Context.NONE);
+        final String accept = "application/json";
+        return service.deleteTestSync(this.getEndpoint(), this.getServiceVersion().getVersion(), testId, accept,
+            requestOptions, Context.NONE);
     }
 
     /**
@@ -2936,8 +2942,9 @@ public final class LoadTestAdministrationClientImpl {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Void>> deleteTestProfileWithResponseAsync(String testProfileId,
         RequestOptions requestOptions) {
+        final String accept = "application/json";
         return FluxUtil.withContext(context -> service.deleteTestProfile(this.getEndpoint(),
-            this.getServiceVersion().getVersion(), testProfileId, requestOptions, context));
+            this.getServiceVersion().getVersion(), testProfileId, accept, requestOptions, context));
     }
 
     /**
@@ -2956,8 +2963,9 @@ public final class LoadTestAdministrationClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> deleteTestProfileWithResponse(String testProfileId, RequestOptions requestOptions) {
+        final String accept = "application/json";
         return service.deleteTestProfileSync(this.getEndpoint(), this.getServiceVersion().getVersion(), testProfileId,
-            requestOptions, Context.NONE);
+            accept, requestOptions, Context.NONE);
     }
 
     /**
diff --git a/sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/implementation/LoadTestRunClientImpl.java b/sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/implementation/LoadTestRunClientImpl.java
index 795e6ea68ab..1b8baee686e 100644
--- a/sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/implementation/LoadTestRunClientImpl.java
+++ b/sdk/loadtesting/azure-developer-loadtesting/src/main/java/com/azure/developer/loadtesting/implementation/LoadTestRunClientImpl.java
@@ -231,7 +231,7 @@ public final class LoadTestRunClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<Void>> deleteTestRun(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testRunId") String testRunId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Delete("/test-runs/{testRunId}")
         @ExpectedResponses({ 204 })
@@ -241,7 +241,7 @@ public final class LoadTestRunClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Response<Void> deleteTestRunSync(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testRunId") String testRunId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Get("/test-runs/{testRunId}/app-components")
         @ExpectedResponses({ 200 })
@@ -485,7 +485,7 @@ public final class LoadTestRunClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Mono<Response<Void>> deleteTestProfileRun(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testProfileRunId") String testProfileRunId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Delete("/test-profile-runs/{testProfileRunId}")
         @ExpectedResponses({ 204 })
@@ -495,7 +495,7 @@ public final class LoadTestRunClientImpl {
         @UnexpectedResponseExceptionType(HttpResponseException.class)
         Response<Void> deleteTestProfileRunSync(@HostParam("endpoint") String endpoint,
             @QueryParam("api-version") String apiVersion, @PathParam("testProfileRunId") String testProfileRunId,
-            RequestOptions requestOptions, Context context);
+            @HeaderParam("Accept") String accept, RequestOptions requestOptions, Context context);
 
         @Get("/test-profile-runs/{testProfileRunId}")
         @ExpectedResponses({ 200 })
@@ -1639,8 +1639,9 @@ public final class LoadTestRunClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Void>> deleteTestRunWithResponseAsync(String testRunId, RequestOptions requestOptions) {
+        final String accept = "application/json";
         return FluxUtil.withContext(context -> service.deleteTestRun(this.getEndpoint(),
-            this.getServiceVersion().getVersion(), testRunId, requestOptions, context));
+            this.getServiceVersion().getVersion(), testRunId, accept, requestOptions, context));
     }
 
     /**
@@ -1659,7 +1660,8 @@ public final class LoadTestRunClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> deleteTestRunWithResponse(String testRunId, RequestOptions requestOptions) {
-        return service.deleteTestRunSync(this.getEndpoint(), this.getServiceVersion().getVersion(), testRunId,
+        final String accept = "application/json";
+        return service.deleteTestRunSync(this.getEndpoint(), this.getServiceVersion().getVersion(), testRunId, accept,
             requestOptions, Context.NONE);
     }
 
@@ -4380,8 +4382,9 @@ public final class LoadTestRunClientImpl {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<Response<Void>> deleteTestProfileRunWithResponseAsync(String testProfileRunId,
         RequestOptions requestOptions) {
+        final String accept = "application/json";
         return FluxUtil.withContext(context -> service.deleteTestProfileRun(this.getEndpoint(),
-            this.getServiceVersion().getVersion(), testProfileRunId, requestOptions, context));
+            this.getServiceVersion().getVersion(), testProfileRunId, accept, requestOptions, context));
     }
 
     /**
@@ -4400,8 +4403,9 @@ public final class LoadTestRunClientImpl {
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Response<Void> deleteTestProfileRunWithResponse(String testProfileRunId, RequestOptions requestOptions) {
+        final String accept = "application/json";
         return service.deleteTestProfileRunSync(this.getEndpoint(), this.getServiceVersion().getVersion(),
-            testProfileRunId, requestOptions, Context.NONE);
+            testProfileRunId, accept, requestOptions, Context.NONE);
     }
 
     /**
-- 
2.50.1.windows.1

